### DISCUSSION TITLE
Can't graph a multiple alignment

### DISCUSSION THREAD

**Discussion by rpestana94** [biotite-dev/biotite#322] [Q&A] [answered]
I'm trying to use biotite to graph multiple alignments but I always get the same error:

 graphics.plot_sequence_logo(ax, alignment)
  File "/home/arcadio/anaconda3/envs/py38/lib/python3.8/site-packages/biotite/sequence/graphics/logo.py", line 87, in plot_sequence_logo
    ha="left", va="bottom", color=colors[j],
IndexError: list index out of range

The file test_few_sample.fasta has 13 sequences, each one about 1270 amino acids length.

```
import biotite.sequence as seq
import biotite.sequence.io.fasta as fasta
import biotite.sequence.graphics as graphics
import biotite.sequence.align as align
import matplotlib.pyplot as plt

fasta_file = fasta.FastaFile.read("test_few_sample.fasta")
all_sequences = fasta.get_sequences(fasta_file)

ids = []
sequences = []

for header, seq_str in fasta_file.items():
    identifier = header.split("|")[-1].split()[0].split(":")[0]
    ids.append(identifier)
    sequences.append(seq.ProteinSequence(seq_str))

matrix = align.SubstitutionMatrix.std_protein_matrix()
alignment, order, tree, distances = align.align_multiple(
    sequences, matrix, gap_penalty=(-10,-1), terminal_penalty=False
)

alignment = alignment[:, order]
ids = [ids[i] for i in order]

fig = plt.figure(figsize=(8.0, 3.0))
ax = fig.add_subplot(111)
graphics.plot_sequence_logo(ax, alignment)
ax.set_xticks([5,10,15,20])
ax.set_xlabel("Residue position")
ax.set_ylabel("Bits")
ax.spines["right"].set_visible(False)
ax.spines["top"].set_visible(False)
fig.tight_layout()

plt.show()

```
I don't know where is the error, any help is appreciated.

**Comment by padix-key** [comment, marked as answer]
You stumbled upon a bug in the color scheme generation, thank you for the report. I will prepare a fix for the next release.
The reason is, that at least one of you protein sequences contains either an `'X'` or a `'*'`.
For now, there are two ways for you to fix this:

1. If the sequence contains `'*'`: These symbols probably do not convey information in the plot. So you can remove them via the `remove_stops()` method. In the context of your code this means: `sequences.append(seq.ProteinSequence(seq_str).remove_stops())`.
2. Otherwise you can fix the line that produces this bug directly in your Biotite installation: Go to https://github.com/biotite-dev/biotite/blob/723c09b03f16d4cdbd7a1bc4a2a9014edacfdaf4/src/biotite/sequence/graphics/colorschemes.py#L46 and replace `len(scheme["colors"])` with `len(alphabet)`.

**Reply by rpestana94** [reply]
Thanks, I used the second approach and works perfect
